source/common/stats/custom_stat_namespaces_impl.cc

```diff
@@ -0,0 +1,25 @@
+#include "source/common/stats/custom_stat_namespaces_impl.h"
+
+namespace Envoy {
+namespace Stats {
+
+bool CustomStatNamespacesImpl::registered(const absl::string_view& name) const {
+  return namespaces_.find(name) != namespaces_.end();
```

Comment:
this should be used only in the main thread - added a comment about it https://github.com/envoyproxy/envoy/pull/17357/commits/9175b650cc30aa55a9a1627604f687a1a2f68b6b#diff-b37967e306f801ad09c7a8fc4d481371f310513ce69fe36fc1b70afd840c180eR18, thank you!

Reply:
See https://github.com/envoyproxy/envoy/blob/ff68b337d038acccb770fcceec57441cf0c740e2/source/common/common/thread.h#L197

Can we add
```
ASSERT(Thread::isMainThread(());
```
You'll need to include `common/common/thread.h`. I think that would ensure someone doesn't add a worker-thread call to this without us knowing it.

Note you might need to use  `!isWorkerThread()` rather than `IsMainThread()` depending on how your tests are structured, as sometimes the test thread is not the main thread, but we might want to be able to call this from the test thread.

Reply:
oh I missed this comment. will do.